### ISSUE TITLE
How to get store in initState life cycle ?

### ISSUE THREAD

**Issue by YDJSZY** [brianegan/flutter_redux#155]
Now, I get store only by StoreConnector, if I want to get store in  initState life cycle, how to do ?

**Comment by Taym95** [comment]
Use onInit:

```
StoreConnector<AppState, AppState>(
    onInit: (store )=> print(store) ,
      converter: (store) => store.state,
      builder: (context, state) => Container();
```

**Comment by miquelbeltran** [comment]
What @YDJSZY is asking is how they can get the `StoreConnector` inside the `initState` call of a `StatefulWidget`.

In my experience, you can't, because it will give you an exception.

The closer you can do is to access the StoreConnector inside `didChangeDependencies`.

What are you trying to achieve? Do you want to dispatch an action when a Widget is created? You can use also `StoreProvider.of(context)` to access the Store and dispatch an Action.

**Comment by Taym95** [comment]
Yes, that what I meant, you can access to the store in `onInit`:

```
StoreConnector<AppState, AppState>(
    onInit: (store )=> _yourInitFunction,
      converter: (store) => store.state,
      builder: (context, state) => Container();
```
```
void _yourInitFunction(Store<AppState> store){
	store.dispatch(MyAction(parameter));
}
```

**Comment by brianegan** [comment]
@YDJSZY Does using `onInit` solve your problem?

**Comment by Taym95** [comment]
> @YDJSZY Does using `onInit` solve your problem? If not, I could make some changes to the lib to allow access in `initState` if need be.

Actually it will give an error [here](https://github.com/brianegan/flutter_redux/blob/1e21e2d4ecd57d1a51be3a2c889e3030c70a2777/lib/flutter_redux.dart#L47) with `inheritFromElement() was called before _OnboardingState.initState() completed`, I would like to work on this issue, but how would you fix it?

**Comment by brianegan** [comment]
@Taym95 Yep, that's exactly right. BUT -- there's an alternative method that would allow this to happen:

`context.ancestorInheritedElementForWidgetOfExactType(type)?.widget`

Now, this method is functionally different from `context.inheritFromWidgetOfExactType`, which should still be the default method used to find the Store from a given context. However, if you need to use the Store from an `initState` function, `ancestorInheritedElementForWidgetOfExactType` is your only option. I could see it working like this:

```dart
@override
void initState() {
  final store = StoreProvider.of<AppState>(context, listen: false);
}
```

So, we'd essentially add a `listen` bool to the `of` method which is set to true by default. We'd implement `of` like so:

```dart
  static Store<S> of<S>(BuildContext context, {bool listen = true}) {
    final type = _typeOf<StoreProvider<S>>();
    final provider = (listen
        ? context.inheritFromWidgetOfExactType(type)
        : context
            .ancestorInheritedElementForWidgetOfExactType(type)
            ?.widget) as StoreProvider<S>;

    if (provider == null) throw StoreProviderError(type);

    return provider._store;
  }
```
